addons/sale_management/views/sale_portal_templates.xml

```diff
@@ -72,7 +72,7 @@
         </xpath>
 
         <xpath expr="//section[@id='details']//div[@id='quote_qty']" position="replace">
-            <t t-if="sale_order.state in ['draft', 'sent'] and line.sale_order_option_ids">
+            <t t-if="sale_order.state in ['draft', 'sent'] and (line.product_id in [option.product_id for option in sale_order.sale_order_option_ids])">
```

Comment:
I didn't test it, but couldn't this be applied here too? 
https://github.com/odoo/odoo/pull/120229#discussion_r1239618899